tests/test_odp_zaius_graphql_api_manager.py

```diff
@@ -0,0 +1,364 @@
+
+            api = ZaiusGraphQLApiManager(logger=mock_logger)
+            api.fetch_segments(api_key=self.api_key,
+                               api_host=self.api_host,
+                               user_key=self.user_key,
+                               user_value=self.user_value,
+                               segments_to_check=[])
+
+        mock_request_post.assert_called_once()
+        mock_logger.error.assert_called_once_with('Audience segments fetch failed (decode error).')
+
+    def test_fetch_qualified_segments__network_error(self):
+        with mock.patch('requests.post',
+                        side_effect=request_exception.ConnectionError('Connection error')) as mock_request_post, \
+                mock.patch('optimizely.logger') as mock_logger:
+
+            api = ZaiusGraphQLApiManager(logger=mock_logger)
+            api.fetch_segments(api_key=self.api_key,
+                               api_host=self.api_host,
+                               user_key=self.user_key,
+                               user_value=self.user_value,
+                               segments_to_check=[])
+
+        mock_request_post.assert_called_once()
+        mock_logger.error.assert_called_once_with('Audience segments fetch failed (network error).')
+        mock_logger.debug.assert_called_once_with('GraphQL download failed: Connection error')
+
+    def test_fetch_qualified_segments__400(self):
+        with mock.patch('requests.post') as mock_request_post, \
+                mock.patch('optimizely.logger') as mock_logger:
+
+            def mock_exception(**kwargs):
+                myresponse = Response()
+                myresponse.status_code = 403
+                raise request_exception.HTTPError(response=myresponse)
+
+            mock_request_post.side_effect = mock_exception
+
+            api = ZaiusGraphQLApiManager(logger=mock_logger)
+            api.fetch_segments(api_key=self.api_key,
+                               api_host=self.api_host,
+                               user_key=self.user_key,
+                               user_value=self.user_value,
+                               segments_to_check=["a", "b", "c"])
+
+        # make sure that fetch_segments() is called (once).
+        # could use assert_called_once_with() but it's not needed,
+        # we already it assert_called_once_with() in test_fetch_qualified_segments__valid_request()
+        mock_request_post.assert_called_once()
+        # assert 403 error log
+        mock_logger.error.assert_called_once_with('Audience segments fetch failed (403).')
+
+    def test_fetch_qualified_segments__500(self):
+        with mock.patch('requests.post') as mock_request_post, \
+                mock.patch('optimizely.logger') as mock_logger:
+
+            def mock_exception(**kwargs):
+                myresponse = Response()
+                myresponse.status_code = 500
+                raise request_exception.HTTPError(response=myresponse)
```

Comment:
If we return a `Response` object from `requests.post` then `raise_for_status` will raise the correct error itself.